dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -387,15 +383,12 @@ public Workflow refresh(@ApiParam(hidden = true) @Auth User user,
 
         // do a full refresh when targeted like this
         // If this point has been reached, then the workflow will be a FULL workflow (and not a STUB)
-        if (workflow.getDescriptorType() == DescriptorLanguage.SERVICE) {
-            workflow.setMode(SERVICE);
-        } else {
+        if (!Objects.equals(workflow.getMode(), DOCKSTORE_YML)) {
             workflow.setMode(WorkflowMode.FULL);
         }
 
         // look for checker workflows to associate with if applicable
-        if (workflow instanceof BioWorkflow && !workflow.isIsChecker() && workflow.getDescriptorType() == CWL
-            || workflow.getDescriptorType() == WDL) {
+        if (workflow instanceof BioWorkflow && !workflow.isIsChecker() && workflow.getDescriptorType() == CWL || workflow.getDescriptorType() == WDL) {
```

Comment:
It was like this, so assume it's OK, but just double checking -- we are not handling Nextflow here on purpose, right?

Reply:
I am pretty sure nextflow does not yet (or at least at the time) have checker workflow implementation.